### PR TITLE
Slack and GH issue integration for scheduled model regression test workflow

### PR DESCRIPTION
**Pull request by HotThoughts**
**Proposed changes**:
Adding a new job in the scheduled Model Regression Test workflow. When model performance drops or test failure: 
- Notify slack
- Open issue with :bug: label
- Move the issue to research's board

Right now the issues for test failure and performance drop are separate. The test failure is notified per configuration, so one issue per configuration. Unlike test failure, performance is evaluated after all configurations is completed. 
In short, the research channel will receive many notifications if multiple configuration test fail, but only one notification for performance drop.

Related to https://github.com/RasaHQ/rasa/issues/8035 and https://github.com/RasaHQ/rasa/issues/7971

An improvement for this issue would be: https://github.com/RasaHQ/rasa/issues/8268

**Current status:**
- [X] Add TF2.3 deterministic mode
- [X] Detect performance drop by matching if the performance description (a table) has anything negative values inside parentheses. 
- [X] Create an issue for it
- [X] Randomly pick a team member to work on the issue
- [X] Add label to the issue
- [X] Put details into the issue body
- [X] Move the issue to the project inbox column as a card
        Example issue: https://github.com/RasaHQ/rasa/issues/8191

        

**TODOs**
- [x] Remove testing lines

### CODE REVIEW

**Comment by HotThoughts** [comment]
@tczekajlo this task is almost there. We need a service account to move the issue to the project inbox column via GitHub REST API.

File: .github/workflows/ci-model-regression-on-schedule.yml [hunk 1]
```diff
@@ -257,6 +261,148 @@ jobs:
...
+          curl \
+            -X POST \
+            -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github.v3+json" \
+            https://api.github.com/repos/$GITHUB_REPOSITORY/issues/${ISSUE_NUMBER}/labels \
+            -d '{"labels":["${{ env.LABEL }}"]}'
+
+      - name: Assign Issue to Project Column
+        run: |
+          ISSUE_ID=$(echo '${{ steps.open_issue.outputs.issue }}' | jq '.id')
```
Comment (tczekajlo) [inline .github/workflows/ci-model-regression-on-schedule.yml:366]:
```suggestion
          ISSUE_ID=$(echo '${{ steps.open_issue.outputs.issue }}' | jq -r '.id')
```

**Comment by HotThoughts** [comment]
@dakshvar22 do you have time 👀? The workflow now added TF deterministic mode, and will send slack notification + create an issue whenever a configuration test failed. Performance drops will be evaluated when all test runs completed, so there will be only one slack notification and one issue for performance drops.

As you mentioned last week multiple test failures should ideally be captured in one issue, I create a new issue #8268 for it. To achieve this, we need a bit more time to investigate.

**Review by dakshvar22 — COMMENTED** [review]
Setting the environment variable looks good to me. I'll hold off from approving before Tomasz can review everything else.
One question -> Do the slack notifications have the link to the github issue also included in them?

File: .github/workflows/ci-model-regression-on-schedule.yml [hunk 2]
```diff
@@ -175,6 +175,9 @@ jobs:
         run: |-
           poetry run rasa --version
 
+          # Run GPUs in a deterministic mode
+          export TF_DETERMINISTIC_OPS=1
```
Comment (tczekajlo) [inline .github/workflows/ci-model-regression-on-schedule.yml:179]: Can we move to the `env` a few lines above?

**Review by tczekajlo — APPROVED** [review]
I've left one small comment

File: .github/workflows/ci-model-regression-on-schedule.yml (hunk 2 above)
Comment (HotThoughts) [reply .github/workflows/ci-model-regression-on-schedule.yml]: Moved!

**Comment by HotThoughts** [comment]
@tczekajlo @dakshvar22 This PR is ready for the second iteration of reviewing. Please ✅ when you have time. 🥰

**Comment by HotThoughts** [comment]
I enabled the auto-merge feature and the branch is automatically merged 😳.

**Pull request merged**